src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,146 @@
+
+        case "delete":
+            String[] delTokens = instruction.split(" ");
+            if (delTokens.length != 2) {
+                throw new DukeException("\t" + "Please key in the correct format of your delete command.");
+            }
+
+            int deletedTaskNumber;
+            try {
+                deletedTaskNumber = Integer.parseInt(delTokens[1]);
+            } catch (NumberFormatException ne) {
+                throw new DukeException("\t" + "Please specify the number of the task that you"
+                        + " wish to delete.");
+            }
+            return new DeleteCommand(deletedTaskNumber);
+
+        case "whatsup":
+            String[] whatsupTokens = instruction.split("/on ");
+            LocalDate queryDate;
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+                queryDate = LocalDate.parse(whatsupTokens[1].trim(), formatter);
+            } catch (DateTimeParseException e) {
+                throw new DukeException("\t" + "Please write the date in this format: "
+                        + "dd-MM-yyyy");
+            } catch (Exception e) {
+                throw new DukeException("\t" + "Please specify the Whatsup command in the right format.");
+            }
+            return new WhatsupCommand(queryDate);
+
+        case "todo":
+            String taskName;
+            try {
+                taskName = instruction.split("todo ")[1];
+                if (taskName.equals("")) {
+                    throw new DukeException("\t" + "Invalid Todo Format.");
+                }
+            } catch (Exception e) {
+                throw new DukeException("\t" + "Please do not leave todo description as empty.");
+            }
+            return new AddCommand("todo", taskName);
+
+        case "event":
+            String[] eventTokens = instruction.split("/at ");
+            String timePeriod;
+            String eventName;
+            try {
+                timePeriod = eventTokens[1];
+                eventName = eventTokens[0].split("event ")[1].trim();
+                if (timePeriod.equals("") || eventName.equals("")) {
+                    throw new DukeException("\t" + "Invalid Event Format.");
+                }
+            } catch (ArrayIndexOutOfBoundsException aiobe) {
+                throw new DukeException("\t" + "Please do not leave the event description (partially) empty.");
+            }
+            return new AddCommand("event", eventName, timePeriod);
+
+        case "deadline":
+            String[] deadlineTokens = instruction.split("/by ");
+            String dueDate;
+            String deadlineName;
+
+            try {
+                dueDate = deadlineTokens[1];
+                deadlineName = deadlineTokens[0].split("deadline ")[1].trim();
+                if (dueDate.equals("") || deadlineName.equals("")) {
+                    throw new DukeException("\t" + "Invalid Deadline Format.");
```

Comment:
Might want to have a specific exception for each kind of error instead of putting all into DukeException. By separating it, the exception name becomes self explanatory as well, that will help in knowing straightaway what kind of exception it is, and how it is being handled.